elephant/spike_train_correlation.py

```diff
@@ -588,64 +600,87 @@ def cross_correlation_histogram(
-        # Assign left and right edges of the cch
         left_edge, right_edge = window[0], window[1]
-        # The mode in which to compute the cch for the speed implementation
+        lags = np.arange(left_edge + t_start_shift,
+                         right_edge + 1 + t_start_shift, dtype=np.int32)
         cch_mode = 'pad'
-    # Case without explicit window parameter
     elif window == 'full':
         # cch computed for all the possible entries
-        # Assign left and right edges of the cch
-        right_edge = binned_st2.num_bins - 1
-        left_edge = - binned_st1.num_bins + 1
+        # ex. 1) lags range: [-6, 9] ms
+        # ex. 2) lags range: [-4, 7] ms
+        # ex. 3) lags range: [-2, 4] ms
+        left_edge = window_min
+        right_edge = window_max
+        lags = np.arange(left_edge + t_start_shift,
+                         right_edge + 1 + t_start_shift, dtype=np.int32)
         cch_mode = window
-        # cch compute only for the entries that completely overlap
     elif window == 'valid':
-        # cch computed only for valid entries
-        # Assign left and right edges of the cch
-        right_edge = max(binned_st2.num_bins - binned_st1.num_bins, 0)
-        left_edge = min(binned_st2.num_bins - binned_st1.num_bins, 0)
+        if binned_st1.num_bins < binned_st2.num_bins:
+            # ex. 1) lags range: [-2, 5] ms
+            # ex. 2) lags range: [1, 2] ms
+            left_edge = (binned_st2.t_start -
+                         binned_st1.t_start) / binsize
+            right_edge = (binned_st2.t_stop -
+                          binned_st1.t_stop) / binsize
+        else:
+            # ex. 3) lags range: [-1, 3] ms
+            left_edge = (binned_st2.t_stop -
+                         binned_st1.t_stop) / binsize
+            right_edge = (binned_st2.t_start -
+                          binned_st1.t_start) / binsize
+        right_edge = int(right_edge.simplified.magnitude)
+        left_edge = int(left_edge.simplified.magnitude)
+        lags = np.arange(left_edge, right_edge + 1, dtype=np.int32)
         cch_mode = window
-    # Check the mode parameter
     else:
         raise ValueError("Invalid window parameter")
+
     if binary:
         binned_st1 = binned_st1.binarize(copy=True)
         binned_st2 = binned_st2.binarize(copy=True)
 
-    cch_builder = CrossCorrHist(binned_st1, binned_st2,
-                                window=(left_edge, right_edge))
+    cch_builder = _CrossCorrHist(binned_st1, binned_st2,
+                                 window=(left_edge, right_edge))
     if method == 'memory':
-        cross_corr = cch_builder.correlate_memory()
+        cross_corr = cch_builder.correlate_memory(cch_mode=cch_mode)
     else:
         cross_corr = cch_builder.correlate_speed(cch_mode=cch_mode)
-    bin_ids = np.arange(left_edge, right_edge + 1)
+
     if border_correction:
         cross_corr = cch_builder.border_correction(cross_corr)
```

Comment:
If I understand this correctly the border correction does not make any sense in the case `window='valid'`. If everything is calculated at full overlap then the same amount of bins is being used for each calculation, so there is no bin loss at the edges. In fact, a correct first order border correction should be zero in this case, so I suspect there is an issue in the calculation of the correction. 
I think I know what is wrong, but in PR reviews on github it is not possible to comment on lines untouched by the PR. Would you prefer to fix that here or rather open a new PR?
In any case there should probably be a warning that setting both `window='valid'` and `border_correction=True` does not make sense.

Reply:
> a correct first order border correction should be zero in this case, so I suspect there is an issue in the calculation of the correction.

I agree, this is also my thought. Till the issue is fixed, I decided to ignore border correction when window mode is `valid`. But let's better create a separate fix for this.

Apart from border correction, have you found any other issues with this PR? I wanted someone to take a look (not a closer one, just conceptional) before merging.

Reply:
Alright! Haven't found any other issues, looks good!